### DISCUSSION TITLE
Increase the size of the .text section in ELF.

### DISCUSSION THREAD

**Discussion by vl17x** [redballoonsecurity/ofrak#346] [Q&A]
Hi !!!
Maybe someone has a good experience with increasing the size of the .text section of an ELF file, share an example..
I need to unlock some functions in program I know how, but it needs more space.

**Comment by EdwardLarson** [comment]
Hello! 

Increasing the .text section is often tricky because simply increasing the size would force moving the sections after it. In OFRAK, we have a the [ElfLoadAlignmentModifier](https://ofrak.com/docs/reference/ofrak/core/elf/load_alignment_modifier.html#ofrak.core.elf.load_alignment_modifier.ElfLoadAlignmentModifier) which will look for unused alignment bytes next to the .text section and add those to the .text section, increasing the size. However this requires that there are unused alignment bytes, which is not always the case.

An alternative approach is the [LiefAddSegmentModifier](https://ofrak.com/docs/reference/ofrak/core/elf/lief_modifier.html#ofrak.core.elf.lief_modifier.LiefAddSegmentModifier), which can add a segment to a binary. This has its own restrictions, because it requires that there is already a `.NOTE` segment in the target ELF which can be replaced by the new segment. An example of using this modifier can be found [here](https://ofrak.com/docs/examples/ex5_binary_extension.html).

Thanks for asking!
